lib/appnexusapi/faraday/raise_http_error.rb

```diff
@@ -31,7 +31,15 @@ def on_complete(response)
           end
 
           return if response.body.empty?
-          body = JSON.parse(response.body).fetch('response', {})
+
+          body =
+            begin
+              JSON.parse(response.body).fetch('response', {})
```

Comment:
I can't tell what `response.body` is returning (because the Faraday docs literally say it returns [Object](http://www.rubydoc.info/gems/faraday/Faraday/Response#body-instance_method)), but if we pull out the `response` (which I'm guessing is specifically the HTTP body of the response), aren't we going to lose the response header information in the positive case, so that we can no longer tell if `body['error_code'] == RATE_EXCEEDED_ERROR` (because it's now nil)?

what was the rationale behind this block?

Reply:
I assume though this isn't any different than the code before, it's doing the same thing except catching the error, which would have errored out prior to this point anyways (basically the previous code had the same issue)

Reply:
derp. I had Github PRs on `Split` viewing for a particular PR yesterday and didn't turn it off. I wrongly assumed due to the whitespace that this was entirely new behavior. nevermind.

Reply:
yeah, so before it didn't allowed anything else but JSON, but the report response is a CSV sent in the body, without any json wrapping, so i rescued the json parse error just to be able to receive the CSV